### PR TITLE
Moved sonar to ci-cd pipeline

### PR DESCRIPTION
**Pull request by LucianoCaravajal**
# JS SDK

## Extra Notes

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -85,3 +85,33 @@ jobs:
           SOURCE_DIR: ./umd
           DEST_DIR: sdk
           ARGS: --acl public-read --follow-symlinks --cache-control max-age=31536000,public
+
```
Comment (melinaurruchua) [inline .github/workflows/ci-cd.yml:88]: Add `fetch-depth: 0` to checkout code for sonar to work

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -85,3 +85,33 @@ jobs:
...
+      - name: SonarQube Scan (Push)
+        if: github.event_name == 'push'
+        uses: SonarSource/sonarcloud-github-action@v1.5
```
Comment (melinaurruchua) [inline .github/workflows/ci-cd.yml:91]: v1.6 is available

File: .github/workflows/ci-cd.yml [hunk 3]
```diff
@@ -85,3 +85,33 @@ jobs:
           SOURCE_DIR: ./umd
           DEST_DIR: sdk
           ARGS: --acl public-read --follow-symlinks --cache-control max-age=31536000,public
+
+      - name: SonarQube Scan (Push)
+        if: github.event_name == 'push'
```
Comment (melinaurruchua) [inline .github/workflows/ci-cd.yml:90]: we can ask sdk team if necessary on pulls too

**Review by melinaurruchua — CHANGES_REQUESTED** [review]

File: .github/workflows/sonar-scan.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      - master
+      - development
+  push:
+    branches:
+      - master
+      - development
+
+permissions:
+  contents: read
+  id-token: write
```
Comment (Israphel) [inline .github/workflows/sonar-scan.yml:14]: not required, you're not assuming roles

Comment (LucianoCaravajal) [reply .github/workflows/sonar-scan.yml]: Removed!

File: .github/workflows/sonar-scan.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+          projectBaseDir: .
+          args: >
+            -Dsonar.host.url=${{ secrets.SONARQUBE_HOST }}
+            -Dsonar.projectName=${{ github.event.repository.name }}
+            -Dsonar.projectKey=${{ github.event.repository.name }}
+            -Dsonar.links.ci="https://github.com/splitio/${{ github.event.repository.name }}/actions"
+            -Dsonar.links.scm="https://github.com/splitio/${{ github.event.repository.name }}"
+      - name: SonarQube Scan (Pull Request)
+        if: github.event_name == 'pull_request'
+        uses: SonarSource/sonarcloud-github-action@v1.5
```
Comment (melinaurruchua) [inline .github/workflows/sonar-scan.yml:52]: v1.6 here too

**Review by melinaurruchua — CHANGES_REQUESTED** [review]

File: .github/workflows/sonar-scan.yml (hunk 5 above)
Comment (LucianoCaravajal) [reply .github/workflows/sonar-scan.yml:52]: I didn't see that!! thanks for pointing it out

**Review by melinaurruchua — APPROVED** [review]

**Pull request merged**